src/main/java/org/sobotics/guttenberg/finders/RelatedAnswersFinder.java

```diff
@@ -77,13 +77,30 @@ public RelatedAnswersFinder(List<Integer> ids) {
                 
                 List<JsonObject> relatedFinal = new ArrayList<JsonObject>();
                 
-                JsonObject relatedAnswers = ApiUtils.getAnswersToQuestionsByIdString(relatedIds, "stackoverflow", prop.getProperty("apikey", ""));
-                //System.out.println(relatedAnswers);
-                for (JsonElement answer : relatedAnswers.get("items").getAsJsonArray()) {
-                    JsonObject answerObject = answer.getAsJsonObject();
-                    relatedFinal.add(answerObject);
-                }
+                boolean hasMore = true;
+                int i = 1;
                 
+                while (hasMore && i <= 3) {
+                	LOGGER.info("Fetch page "+i);
+                	JsonObject relatedAnswers = ApiUtils.getAnswersToQuestionsByIdString(relatedIds, "stackoverflow", prop.getProperty("apikey", ""));
+                    //System.out.println(relatedAnswers);
+                    
+                    for (JsonElement answer : relatedAnswers.get("items").getAsJsonArray()) {
+                        JsonObject answerObject = answer.getAsJsonObject();
+                        relatedFinal.add(answerObject);
+                    }
+                    
+                    JsonElement hasMoreElement = relatedAnswers.get("has_more");
+                    
+                    if (hasMoreElement != null && hasMoreElement.getAsBoolean()) {
+                    	hasMore = true;
+                    } else {
+                    	hasMore = false;
+                    }
```

Comment:
Here you can simply do `hasMore = hasMoreElement != null && hasMoreElement.getAsBoolean()`. Clearer. Probably you don't even need the `hasMore` anymore even.